### ISSUE TITLE
Memory stress is consuming full CPU core

### ISSUE THREAD

**Issue by anuragpaliwal80** [chaos-mesh/chaos-mesh#1964]
## Bug Report

**What version of Kubernetes are you using?**

`1.19.4`

**What version of Chaos Mesh are you using?**

`1.1.3`

**What did you do?**

Run memory stress tests:
```
  stressors:
    memory:
      workers: 1
      size: '512MB'
```

**What did you expect to see?**
The CPU should not go up.

**What did you see instead?**
It consume 1 full core CPU and if we increase number of worker, it will consume those many CPU core.

**Output of chaosctl**

**Comment by AsterNighT** [comment]
This is a known issue of `stress-ng`. It is caused by frequently write to the memory allocated by `stress-ng`. We are considering, though, using another tool to apply memory stress. Is that it? @YangKeao

**Comment by YangKeao** [comment]
> This is a known issue of `stress-ng`. It is caused by frequently write to the memory allocated by `stress-ng`. We are considering, though, using another tool to apply memory stress. Is that it? @YangKeao 

Yes. But before doing that, I think we should define the behaviour of "Memory Stress".

Stressing the disk is to read/write the disk. Therefore, defining the "Memory Stress" as writing/reading too much to squeeze the bandwidth of memory cannot be asserted as wrong 🤦‍♂️

But I also believe we should provide a way to eat RAM without costing too much CPU🍻 (and a new tool other than stress-ng is needed)

**Comment by anuragpaliwal80** [comment]
FYI... The fix is still pending.

**Comment by WLBF** [comment]
@YangKeao  I think a simple program that allocate specific amount  of memory and fill these memory with random bytes to force os allocate same amount of physical memory will do the work.

**Comment by cwen0** [comment]
> @YangKeao I think a simple program that allocate specific amount of memory and fill these memory with random bytes to force os allocate same amount of physical memory will do the work.

Yes, @FingerLeader has developed a simple program to fix this issue in #2383.

**Comment by YangKeao** [comment]
@WLBF Yes. https://github.com/chaos-mesh/chaos-mesh/pull/2623 should have fixed this issue.
